File: rpg-frontend/src/types/leaderboard.ts
export type LeaderboardType = "level" | "pve_wins" | "arena_wins" | "coins";

export interface LeaderboardHeroOut {
  name: string;
  level: number;
}

export interface LeaderboardEntryOut {
  rank: number;
  user_id: number;
  username: string | null;
  hero: LeaderboardHeroOut | null;
  value: number;
}

export interface LeaderboardOut {
  type: string;
  entries: LeaderboardEntryOut[];
  total: number;
  limit: number;
  offset: number;
  my_rank: number | null;
  my_value: number;
}

File: rpg-backend/app/schemas/leaderboard.py
from pydantic import BaseModel


class LeaderboardHeroOut(BaseModel):
    name: str
    level: int


class LeaderboardEntryOut(BaseModel):
    rank: int
    user_id: int
    username: str | None
    hero: LeaderboardHeroOut | None
    value: int


class LeaderboardOut(BaseModel):
    type: str
    entries: list[LeaderboardEntryOut]
    total: int
    limit: int
    offset: int
    my_rank: int | None
    my_value: int
